src/main/java/com/facebook/openwifirrm/modules/ModelerUtils.java

```diff
@@ -381,4 +399,269 @@ public static Map<String, Map<String, WifiScanEntry>> getAggregatedWifiScans(
 		}
 		return aggregatedWifiScans;
 	}
+
+	/** This method converts the agggregated map to a Json String.
+	 *
+	 * @param aggregatedMap 	Aggregated map
+	 * @param aggType 			Determines how to aggregate the statistics.
+	 * @return String			A Json String representation of the aggregated map.
+	 */
+	public static String aggregatedStatesToJson(
+		Map<String, Map<String, Map<String, Map<JsonNode, Map<String, MeanAggregator>>>>> aggregatedMap,
```

Comment:
😬 

it makes sense but a few things I don't like about this:
- too nested
- `String` doesn't give enough information about what kind of string it actually is

I won't block this because I can't think of a better representation off the top of my head though

Reply:
A nested/chained keys like that also cost much hash lookup time. Based on whether we really want to chain these keys or just want a combination of the keys, we can use different methods. I'd suggest a class with those keys and implement `hashCode` and `equals`, and then use that class as the key if a chained set of keys is not really needed.

Reply:
Thanks for comments! I will wait for more inputs from the system team

Reply:
I am going to think about this too, I haven't delved into the details yet...

Reply:
So I think the nested map can be flatten a little bit. A (bssid, station)-pair defines a link. 
For example, 
serial_number: {
     association: {
          bssid: xxxx,
          station: yyyy,
          key: {
               channel: z, 
               channel_width: e,
               tx_power: r
          },
          stats: {
               mcs: [d1, d2, d3],
               rssi: [f1, f2, f3]
          }
     }
}
Let me know what you guys think about this. Thanks.